flink-core/src/main/java/org/apache/flink/core/fs/LimitedConnectionsFileSystem.java

```diff
@@ -88,6 +90,12 @@
 	/** The nanoseconds that a stream may spend not writing any bytes before it is closed as inactive. */
 	private final long streamInactivityTimeoutNanos;
 
+	/** Rate limiter of incoming bytes for this filesystem. */
+	private final RateLimiter inputRateLimiter;
```

Comment:
Instead of nullable field, please use `Optional<RateLimiter>` on both fields. annotating `@Nullable` wouldn't suffice, since unfortunately we do not enforce such annotation and `Optional` is always enforced.